pkg/sdk/go/things_test.go

```diff
@@ -646,7 +646,7 @@ func TestThing(t *testing.T) {
 			response: sdk.Thing{},
 			token:    validToken,
 			thingID:  wrongID,
-			err:      errors.NewSDKErrorWithStatus(errors.Wrap(repoerr.ErrNotFound, repoerr.ErrNotFound), http.StatusNotFound),
+			err:      errors.NewSDKErrorWithStatus(errors.Wrap(svcerr.ErrViewEntity, svcerr.ErrViewEntity), http.StatusUnprocessableEntity),
```

Comment:
Why do we wrap the error with the same error?

Reply:
This is because the API response from the service contains an `error` and `message` i.e

```json
{"error":"failed to perform authorization over the entity","message":"failed to perform authorization over the entity"}
```

When checking `SDKError` we wrap both the error and the message https://github.com/absmach/magistrala/blob/5c62e339c1ab63aa1b1697ea12e39e7ac42aaf43/pkg/errors/sdk_errors.go#L122

Reply:
This indicates poor error handling, but it is not a subject of this PR. Some of the error handling tickets should take care of it.